### PR TITLE
RUN-1592 + RUN-1591 + RUN-955: fix issues related to events and event previews + OnConsoleMessage JS exception + crash on compile failure

### PR DESCRIPTION
**Pull request by Domiii**
* Paired w/ https://github.com/replayio/chromium-v8/pull/121
* → Fix issues related to events and event previews - https://linear.app/replay/issue/RUN-1592/fix-issues-related-to-events-and-event-previews
* → Also: Crash on Replay script compile failure - https://linear.app/replay/issue/RUN-955
* Tests [here](https://github.com/RecordReplay/build-test-dashboard/blob/master/153b0f17-eea6-4dd0-8040-545ed55bd21c): 87 vs. 11 (total 100)
  * Honeycomb [here](https://ui.honeycomb.io/replay/datasets/backend/result/HP1AVnA8nDG): More command errors than usual 🤔 
* Fixed several bugs [here](https://github.com/replayio/chromium/pull/495/commits/bad74243d6a2e03ff10c528613af31b1edb4f16f)
* Tested again [here](https://github.com/RecordReplay/build-test-dashboard/blob/master/c40afe96-7dcc-4bf1-b980-5d2bb75e3766): 137 vs. 13 (150 total)
  * Honeycomb crashes [here](https://ui.honeycomb.io/replay/datasets/backend/result/wVxEAowDxUT?tab=raw):
    * 32 x command, 13 x `ControllerFinished`
    * Quite a few crashes, but most of them are cross-origin-related:
      * 26 segfault in `V8Window::NamedPropertyGetterCustom` in `v8_window::CrossOriginNamedGetterCallback` when trying to evaluate `isArrayLike` in `ValueMirror::create`
        * Maybe this is fall-out from [RUN-1315](https://linear.app/replay/issue/RUN-1315/very-bad-command-performance-getallframes-wandb) but it is certainly unrelated to this work.
        * Filed a follow-up issue instead: [RUN-1619](https://linear.app/replay/issue/RUN-1619/segfault-in-v8windownamedpropertygettercustom-during)
      * 6 JS errors involving the pattern `"Error: Blocked a frame with origin \"https://www.huffpost.com\" from accessing a cross-origin frame."`

### CODE REVIEW

**Review by bhackett1024 — APPROVED** [review]

**Comment by Domiii** [comment]
@bhackett1024 I need a second review. After running tests:

* I fixed several JS errors in [this commit](https://github.com/replayio/chromium/pull/495/commits/bad74243d6a2e03ff10c528613af31b1edb4f16f).
* I then re-ran tests and detailed my findings in [the description](https://github.com/replayio/chromium/pull/495#issue-1646129945) of this PR.

**Review by bhackett1024 — APPROVED** [review]
@Domiii in the future, can you wait until tests look good before asking for a review with a complex PR like this?

**Pull request merged**
